src/http.ts

```diff
@@ -458,6 +479,45 @@ export async function getReport(options: GetReportOptions): Promise<Result<Uploa
   return generateError<ReportErrorCodes>(res.errorCode, REPORT_ERROR_MESSAGES, 'getReport', res.error?.message);
 }
 
-export function getVerifyCallbackUrl(authHost: string): string {
-  return `${authHost}/api/verify/callback`;
+/**
+ * Trigger an SCM-based test with reporting.
+ */
+export async function initScmReport(options: ScmUploadReportOptions): Promise<Result<string, ReportErrorCodes>> {
+  const config: Payload = {
+    headers: {
+      ...commonHttpHeaders(options),
+    },
+    url: `${options.baseURL}/test`,
+    method: 'post',
+    body: {
+      workflowData: {
+        projectId: options.projectId,
+        commitHash: options.commitId,
+      },
+      ...pick(options, ['severity', 'prioritized', 'analysisContext']),
+    },
+  };
+
+  const res = await makeRequest<InitScmUploadResponseDto>(config);
+  if (res.success) return { type: 'success', value: res.body.testId };
+  return generateError<ReportErrorCodes>(res.errorCode, REPORT_ERROR_MESSAGES, 'initReport');
```

Comment:
q: Are the user-facing errors introduced with the report flow going to be the same as before, or are we planning on adding more or changing them? Just wanted to make sure this is something that doesn't get forgotten, but it's not that urgent IMO

Also applies to line 522

Reply:
Good question, I went back and forth a bit on this, and considered adding specific "initScmReport" and "getScmReport" errors, but since right now both flows should have the same user-facing errors I opted against duplication.
Depending on what our initial customer tests will surface, and as we find some failure cases that maybe should be propagated to users, we can expand the common cases or separate the 2 flows eventually if they'll have specific enough errors.